cpp/src/parquet/column_reader.cc

```diff
@@ -1478,13 +1478,21 @@ class TypedRecordReader : public ColumnReaderImplBase<DType>,
     int64_t num_decoded = this->current_decoder_->DecodeSpaced(
         ValuesHead<T>(), static_cast<int>(values_with_nulls),
         static_cast<int>(null_count), valid_bits, valid_bits_offset);
-    DCHECK_EQ(num_decoded, values_with_nulls);
+    if (ARROW_PREDICT_FALSE(num_decoded != values_with_nulls)) {
+      ParquetException::EofException("Decoded values " + std::to_string(num_decoded) +
+                                     " does not match expected " +
+                                     std::to_string(values_with_nulls));
+    }
```

Comment:
Can you factor out the check and error message in a convenience function or method?

Reply:
Done